src/App.tsx

```diff
@@ -15,48 +141,127 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+export const defSorting = {
+  method: '',
+  isReversed: false,
+};
+
+enum Metods {
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+type SortingOptions = {
+  method: string;
+  isReversed: boolean;
+};
+
+function preparedGoods(
+  goods: string[],
+  { method, isReversed }: SortingOptions,
+): string[] {
+  const getGoods = [...goods];
+
+  if (method === Metods.Alphabet) {
+    getGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (method === Metods.Length) {
+    getGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (isReversed) {
+    getGoods.reverse();
+  }
+
+  return getGoods;
+}
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState(defSorting);
+  const goods = preparedGoods(goodsFromServer, sortBy);
+
+  const handleResetSorting = () => {
+    setSortBy({
+      method: '',
+      isReversed: false,
+    });
+  };
+
+  const handleReverse = () => {
+    setSortBy(
+      sortBy.isReversed
+        ? { ...sortBy, isReversed: false }
+        : { ...sortBy, isReversed: true },
+    );
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortBy({ ...sortBy, method: Metods.Alphabet })}
+          className={
+            classNames(
+              'button',
+              'is-info',
+              {
+                'is-light': sortBy.method !== 'alphabet',
```

Comment:
Good tip, please change
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/54936044/18aeb575-9ad8-4f16-836f-30e29e3a3d83)